MEDMI_Data_Download/met_extraction_script.py

```diff
@@ -48,107 +163,237 @@ def _extract_data(self, dict, settings, extra_datasets=[]):
+        self._cols_specific = [MetExtractorTemp.MEASUREMENT_NAME]
+        self._filename = '{}/temp{}{}.csv'.format(self._out_dir, '{}', '{}')
 
 
-class MetExtractorWind(MetExtractor):
+class MetExtractorRelativeHumidity(MetExtractor):
+    SOURCE_REFERENCE = 'midas.weather_hrly_ob.rltv_hum'
+    MEASUREMENT_NAME = 'rel_hum'
 
-    def __init__(self, outfile_dir):
-        super(MetExtractorWind, self).__init__(outfile_dir)
-        self.source_reference = 'midas.wind_mean_ob.mean_wind_speed 1'
-        self.complex_wind_type = True
-        self.head_string = 'Wind speed and direction hourly data for date range: {} to {}\n'
+    def __init__(self, out_dir=MetExtractor.DEFAULT_OUT_DIR, verbose=MetExtractor.DEFAULT_VERBOSE):
+        super(MetExtractorRelativeHumidity, self).__init__(out_dir, verbose)
+        self._head_string = 'Relative Humidity {}  data for date range: {} to {}\n'
+        self._cols_specific = [MetExtractorRelativeHumidity.MEASUREMENT_NAME]
+        self._filename = '{}/rel_hum{}{}.csv'.format(self._out_dir, '{}', '{}')
 
-    def extract_data(self, date_range, latitude_range, longitude_range, outfile_suffix):
-        self.outfile_suffix = outfile_suffix
-        self.filename = '{}/wind_{}.csv'.format(new_dir, outfile_suffix)
-        print('extracting {}'.format(self.head_string.format(date_range[0], date_range[1])))
-        dict = super(MetExtractorWind, self)._create_dict_base(date_range, latitude_range, longitude_range)
-        dict.update({'Source reference': self.source_reference, 'Complex wind type': self.complex_wind_type})
 
-        settings = {    'fname': self.filename,
-                        'headstring': self.head_string.format(date_range[0], date_range[1]),
-                        'columnstring': self.cols_base.format('windspeed,winddir')}
+class MetExtractorStationPressure(MetExtractor):
+    SOURCE_REFERENCE = 'midas.weather_hrly_ob.stn_pres'
+    MEASUREMENT_NAME = 'pressure'
 
-        self._extract_data(dict, settings)
+    def __init__(self, out_dir=MetExtractor.DEFAULT_OUT_DIR, verbose=MetExtractor.DEFAULT_VERBOSE):
+        super(MetExtractorStationPressure, self).__init__(out_dir, verbose)
+        self._head_string = 'Station Pressure {} data for date range: {} to {}\n'
+        self._cols_specific = [MetExtractorStationPressure.MEASUREMENT_NAME]
+        self._filename = '{}/pressure{}{}.csv'.format(self._out_dir, '{}', '{}')
+
+
+
+class MetExtractorDewpoint(MetExtractor):
+    SOURCE_REFERENCE = 'midas.weather_hrly_ob.dewpoint'
+    MEASUREMENT_NAME = 'dewpoint'
+
+    def __init__(self, out_dir=MetExtractor.DEFAULT_OUT_DIR, verbose=MetExtractor.DEFAULT_VERBOSE):
+        super(MetExtractorDewpoint, self).__init__(out_dir, verbose)
+        self._head_string = 'Dewpoint hourly data {} for date range: {} to {}\n'
+        self._cols_specific = [MetExtractorDewpoint.MEASUREMENT_NAME]
+        self._filename = '{}/wet_bulb{}{}.csv'.format(self._out_dir, '{}', '{}')
```

Comment:
'wetbulb' here is a hangover from when I was pulling out the wetbulb temperature (which is different to the dewpoint temperature). The wetbulb temperature can be used to calculate the relative humidity, but the equation required takes both temperature and pressure as inputs too (if I recall correctly) - and in the dataset we have some relative humidity data where there is no pressure reading (so I don't think this was the equation used by the met office). So I switched to extracting the dewpoint temperature instead (which can be used to calculate RH with only temperature, but no pressure, data), but forgot to change the name of the output file too :/

tl:dr - could you change the output file name to 'dewpoint' instead?